app/controllers/ingest_from_ftp_controller.rb

```diff
@@ -32,12 +41,40 @@ def view_status
     @status_results = statuses.sort.to_h
   end
 
+  def delete_packages
+    # need to instantiate source again for redirect
+    source
+    selected_filenames = params[:selected_filenames]
+    if selected_filenames.blank?
+      flash[:alert] = 'No packages were chosen'
+      redirect_to ingest_from_ftp_path(source: @source)
+      return
+    end
+    package_paths = list_package_files
+    package_paths.each do |package_path|
+      if selected_filenames.any? { |filename| package_path.include?(filename) }
+        File.delete(package_path)
+      end
+    end
+    redirect_to ingest_from_ftp_path(source: @source)
+  end
+
   private
 
   def list_package_files
     Dir[File.join(storage_base_path, '*.zip')]
   end
 
+  def list_selected_package_paths(selected_filenames)
+    selected_package_paths = []
+    list_package_files.each do |package_path|
+      if selected_filenames.any? { |filename| package_path.include?(filename) }
```

Comment:
```suggestion
      if selected_filenames.any? { |filename| File.basename(package_path) == filename }
```
since we're only concerned with the filename part and it'd eliminate the very small chance of filename matching only part of the path (either in the parent directories or a partial filename match).